### ISSUE TITLE
About "constructorInvocation" in KoltinParser.g4

### ISSUE THREAD

**Issue by lbtrace** [JetBrains/kotlin-spec#118]
Hi, I am Android developer from China, I using ANTLR4(using KotlinParser.g4 version v1.8) as kotlin AST library. And i find some confusion about "constructorInvocation".

- Kotlin Source Code
```kotlin
class AdAvatarElement : DispatchBaseElement<AvatarAndFollowElementView, AvatarAndFollowViewModel,
    FollowEventBus, RightActionBarGroupEventBus, AdSlidePageConfig, AdSlidePageCallerContext>
  (RightActionBarGroupBizType.AVATAR_AND_FOLLOW, null) {
    ......
}
```
- According KotlinParser.g4 "constructorInvocation" section, it is illegal，right? because ">"(end of userType) And "("(begin of valueArguments) must be same line,  between `userType` and `valueArguments` NOT contains `NL*`
```g4
constructorInvocation
    : userType valueArguments
    ;
```
- ANTLR4 parse this source fail, but Android Studio parse it OK
- My Questions: constructorInvocation of above source code is illegal? Why Android Studio can parse it? @ice-phoenix can you help explain it? CC @drieks

**Comment by belyaev-mikhail** [comment]
Seems like a bug in the grammar to me

**Comment by lbtrace** [comment]
Hi @KvanTTT what do you think about this issue?

**Comment by KvanTTT** [comment]
I think the same: it looks like a bug in the grammar.

**Comment by lbtrace** [comment]
Thanks for replay @belyaev-mikhail @KvanTTT 
So, maybe we should modify `constructorInvocation` like this:
```g4
constructorInvocation
    : userType NL* valueArguments
    ;
```
Do you know who can determine these things?

**Comment by ice-phoenix** [comment]
Yeah, that's definitely a bug, thanks for bringing it to our attention! We'll do our best to fix it in the spec 1.9 release.

**Comment by drieks** [comment]
Hi, I think Android Studio does not use the ANTLR-Grammar for parsing. We have a copy of KotlinParser.g4 here in this repository, so it is possible to fix it here without waiting for upstream changes.
Sadly I still have to setup a new build system :(

**Comment by drieks** [comment]
Hi @lbtrace, Kotlin Studio is not using `KotlinParser.g4`. In kotlinx.ast, we have own copy of this file. But sadly, I still have to fix the build system. If you can use a local build version for kotlinx.ast, you can fix it locally:

https://github.com/kotlinx/ast/blob/master/grammar-kotlin-parser-common/src/commonAntlr/antlr/KotlinParser.g4#L97

```
constructorInvocation
    : userType NL* valueArguments
    ;
```

**Comment by lbtrace** [comment]
Yes, I will fix it locally, thanks for your suggestion

**Comment by ice-phoenix** [comment]
Fixed in a94315649ae44453d6e0a639d4de1a586c8176e2, soon to be released in version 1.9
